### ISSUE TITLE
Reduce conda release frequency?

### ISSUE THREAD

**Issue by mara004** [bblanchon/pdfium-binaries#146]
anaconda.org has a storage limit of 3 GiB, which seems fairly narrow. (You can check status in the Storage tab of your account settings.)
In that light, maybe weekly releases are too frequent? Would it be possible to publish the conda packages only every other week, or once a month, so we don't need to be afraid of running into the limit?

**Comment by bblanchon** [comment]
Lowering the frequency would only postpone the issue.
I think we should discard old versions instead.

**Comment by mara004** [comment]
> I think we should discard old versions instead.

Sounds dangerous? I believe this would threaten significant breakage in pypdfium2, like breaking an entire older major version of the helpers (due to the version pinning and bounding that is necessary to achieve inherent ABI/API safety).
(As an unrelated example, PyPI has a big warning against deleting older version.)

I don't say it's not a possibility, but we'd have to be very careful in what versions to delete, and it would perhaps still risk breaking some callers of pypdfium2.

**Comment by mara004** [comment]
@bblanchon I was wondering if you have reached a resolution on this yet?
Would you be willing to take a PR to run conda publishing only once per month?
It should not affect the other builds, of course.

I believe this would be fairly relevant to achieve a sustainable packaging concept.

**Comment by bblanchon** [comment]
I didn't do anything yet because:

1. There is no rush
2. I'm tired of all the extra work for Anaconda
3. Should we do a monthly release for NuGet too
4. What would people say (hint: people rarely thank) if versions in NuGet/Anaconda differ from the ones on GitHub?
5. It's possible that Ananconda changes the limits before we reach them.
6. They could also make an exception for us.

**Comment by mara004** [comment]
On the other hand, should pdfium ever decide to get rid of AGG and force Skia on embedders, we'd be in size trouble anyway, because Skia is big.
Although they actively pursue Skia, it seems like even Google itself has [binary size concerns](https://crbug.com/pdfium/2088#c4), though.

**Comment by sungaila** [comment]
Personally I don't need weekly NuGet releases. I update the PDFium libs each time I release a new version which happens maybe every few months. You can tell which versions I had pinned by [the download counter of the Win32 packages](https://www.nuget.org/packages/bblanchon.PDFium.Win32#versions-body-tab). 😄

@bblanchon Your work is very much appreciated! I really like your friendly yet productive attitude when discussing issues and pull request.

**Comment by mara004** [comment]
It's a bit vague, but https://github.com/NuGet/Home/issues/6208#issuecomment-346171972 indicates to me that there might just be no disk usage limit on NuGet?
@sungaila Do you have any definite information on this?

**Comment by mara004** [comment]
Thanks for the surprise!

FWIW pypdfium2 does not depend on any releases < 120.0.6097.0, and I doubt if anyone else does, so I wouldn't object to deleting these if needed or desired.

**Comment by mara004** [comment]
In theory we could also consider excluding platforms where there will probably be no actual demand on conda (win-32, linux-32, linux-armv7l ?), but I don't want to trouble you any further. Just an idea e.g. in case pdfium gets bigger or something.

**Comment by bblanchon** [comment]
Indeed, their download page doesn't offer any 32-bit option.
